### DISCUSSION TITLE
DPL: "Inverter is behind power meter" option

### DISCUSSION THREAD

**Discussion by greymda** [hoylabs/OpenDTU-OnBattery#892] [General]
Hello, in wiki there's this interesting setting/statement that confuses me a bit: "Inverter is behind power meter: Select this if your inverter power is measured by the power meter. This is typically the case."

Can someone with knowledge elaborate on this, please?

If this helps, my particular situation is a Shelly EM with 2 clamps for measuring energy.
One clamp is mounted on the initial Line wire that comes into the house, immediately after the energy monitor of the energy provider - so it should monitor the house consumption, I'd say.

Now the second clamp is mounted to a Line wire after the plug the inverter is plugged into, so it shows how much energy the panels are producing at the current moment.

I draw a bit, apologies for my mastery on this field.

<img width="991" alt="image" src="https://github.com/helgeerbe/OpenDTU-OnBattery/assets/9991596/cbe7035e-aa77-4b55-a055-467cedb10712">

In my case: should I enable or disable this option? What it means and can you help with some cases for one or another and maybe we'll update the wiki page as well for future references. 

Cheers

**Comment by spcqike** [comment]
You should enable it and only use the power value from clamp 1.

**Reply by greymda** [reply]
@spcqike what would be then a use case for this option disabled?

**Reply by spcqike** [reply]
Eg if you have a Shelly or something for one or more devices. And you only want to produce energy for these dwvices. 

Or if you live in an apartment and don’t have access to or can’t install a central power meter, but you still want to produce energy out of a battery for some devices. Like Anker and Zendure do. You can setup smart plugs and produce as much as you need. Of course you can’t compensate your total power need with this kind of setup.

**Reply by greymda** [reply]
still a bit confusing the wiki article, to be honest. but from my standpoint, i’ll enable the setting (currently disabled) and only use the power value from clamp 1 (as i do now). thank you!

**Reply by greymda** [reply]
@spcqike  i only now realized it's called different in UI. we are taking about the same thing, correct?
<img width="1418" alt="image" src="https://github.com/helgeerbe/OpenDTU-OnBattery/assets/9991596/e3f2a11c-2b07-421e-82f8-a598899dc966">

**Reply by Manos1966** [reply]
> Inverter is behind power meter: Select this if your inverter power is measured by the power meter. This is typically the case.

Ich interpretiere es so:
Wechselrichter ist hinter Leistungsmesser: Wählen Sie diese Option, wenn die Leistung Ihres Wechselrichters vom Stromzaehler gemessen wird. Dies ist in der Regel der Fall.

NETZ -> Stromzaehler -> Wechselrichter, Haus, Wasserkocher, usw usw...

Wenn der Wechselrichter z.B. 500W produziert, dann werden diese 500W vom Stromzaehler "gesehen" und mitberechnet, also die 500W Leistung deines Wechselrichters wird vom Stromzaehler gemessen.

**Reply by greymda** [reply]
@Manos1966 so in your opinion in my case- should i enable or disable it?

**Reply by spcqike** [reply]
> I only now realized it's called different in UI

AFAIK it was renamed recently.

**Comment by Manos1966** [comment]
> so in your opinion in my case- should i enable or disable it?

Well: You have a Shelly 3EM, which in fact is **three separate "power meters"**: Clamp1, Clamp2 and Clamp3

And you have a fourth "power meter", the Company Energy Monitor. Which I presume you do not have a Data Transfer system to send measurements from this  power meter to your OpenDTU.

Now, the 100EUR question: In your "Powermeter" setup, which clamp/clamps have you defined ?

**Reply by greymda** [reply]
@Manos1966  my unit is a EM (without 3), as 99% households here are on 1 -phase. Anyhow these are my settings, i monitor the clamp that is on the Line from the Energy Meter to the household:
![IMG_5515](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/9991596/ccee0c62-feec-4034-91d5-db37b4130f9d)

**Reply by Manos1966** [reply]
> I monitor the clamp that is on the Line from the Energy Meter (Company Energy Monitor) to the household:

Excellent! 👍 

So, when your Inverter is producing for example 500W of electricity, will the clamp (that is on the Line from the Energy Meter) notice the difference?
For example: 
- Inverter is switched off
- Your house consumes 500W
-  Company Energy Monitor delivers 500W 
- Your clamp (that is on the Line...) reports **500W**

- Now your inverter becomes active and starts producing 400W
- Your house still consumes 500W
- Company Energy Monitor now delivers 100W (the other 400W being delivered by the inverter)
- Your clamp (that is on the Line...) reports now? **100W**

So, this means that, when your inverter delivers energy, the later will be "noticed" by your clamp
Thus, "inverter power is measured by the clamp (i.e. power meter)" is true

Therefore, as spcqike said before, you need to activate it.

I understand the sentence "Inverter is behind power meter: Select this if your inverter power is measured by the power meter. This is typically the case" may be confusing.
How would you best describe it in an understandable way?

**Reply by greymda** [reply]
@Manos1966 

> * Your clamp (that is on the Line...) reports now? **100W**

wouldn’t this mean that the DPL now sets the limit to 100W to produce and hence household start importing from the grid?

**Reply by spcqike** [reply]
That’s what „inverter is behind powermeter“ is for. If it’s not checked, the inverter limit will be set to your power meter value. If it’s checked, the limit will be its actual ~~limit~~ AC output + powermeter reading. 

In the example above it will be 400+100 so 500W.

2024-04-19 12:12 edited by schlimmchen

**Comment by genmad** [comment]
> 
>   How would you best describe it in an understandable way?

Maybe Powermeter: and then a dropdown box with entries: 
1. Measures Consumer and Inverter ( typical case)
2. Measures only some Consumers and not the inverter ( rare case)

Maybe another approach in discribing it could help. Right now you don‘t know what the other case is with the toggle button, I guess.

**Reply by genmad** [reply]
I forgot to mention that this is a  suggestion for the ui to replace the toggle button.

**Reply by schlimmchen** [reply]
To be honest, I don't think this adds clarity, but makes the implementation much more nasty.

I thought hard about the new label text, and I thought I did a good job. Seems there is still more to do. The item has no tooltip with a hint. We could add such a hint. Let's discuss what the hint should read:

"Enable this option if the power meter reading is reduced by the inverter output. This is typically true."

What I would really like to do is remove the option entirely and wait if people do complain that it is missing. Who connects the inverter between the meter and the grid?

**Reply by Manos1966** [reply]
> What I would really like to do is remove the option entirely and wait if people do complain that it is missing. Who connects the inverter between the meter and the grid?

Only people with particular needs, who know what they are doing 👍 

We are trying to explain to people with limited "Know how", a very particular function used by specialists.

On this line of thought, maybe it is easier to:
- add a hint
- "Leave this option active unless you have particular needs "(and you know what you are doing)"

**Reply by spcqike** [reply]
@schlimmchen 
> Who connects the inverter between the meter and the grid?

That’s not the only use case. A probably more common use case is, that you don’t have access to your grid provider energy meter and/or you can’t install a Shelly or whatsoever. Like many people who live in an apartment. The power meter is locked up somewhere and you have to ask someone (Hausmeister) to get your reading. 

Every Anker or Zendur system can handle this situation. You have like 2-3 smart plugs in your flat, and your PV system matches just these few devices. 

IMO it would be a step back if we remove this function in opendtu-onBattery. I don’t know whether or not people actually use it, but that’s probably the most (if not only) useful setup for this checkbox. 

In my previous flat I also would have had just this option. (If a had my HM-600 in the 18.floor already) I just hasn’t permanent access to my mains meter (this was somewhere in a room on the floor)  and wasn’t allowed to install a 3EM in my distribution box in my flat.

**Reply by Manos1966** [reply]
@spcqike Thanks for providing me with the text I need to add to the Wiki 😉

**Comment by Manos1966** [comment]
Currently we have this explanation in Wiki: https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Dynamic-Power-Limiter-%28Nulleinspeisung%29-%E2%80%93-MIT-BATTERIE

**Comment by greymda** [comment]
> "Enable this option if the power meter reading is reduced by the inverter output. This is typically true."

i think this text makes it crystal clear! at least for me, as i was the one initially confused lol

**Reply by schlimmchen** [reply]
FYI: edf493bc6d41880e55c17f79b6e40d59f1c798f5 adds a respective hint (tooltip) to the DPL setting in the Web UI.

**Comment by greymda** [comment]
can you update the wiki page with the text above?

**Reply by Manos1966** [reply]
already done 2 days ago.... 😇

**Reply by greymda** [reply]
@Manos1966  here? https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Dynamic-Power-Limiter

**Reply by Manos1966** [reply]
Ofcourse. Maybe you need to refresh the page inb your browser
![Capture](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/129660360/fc33f8c2-810c-4d33-a631-5b21546e4f92)


EDIT: Oh, I see what you mean, there is another mention before, I see! I guess it is time to put the long(er) English/German explanation in place. Thanks for noticing that 👍
